lib/src/sort.dart

```diff
@@ -0,0 +1,213 @@
+            isAscending: isAscending);
+        _inputs = swap.swapped;
+      }
+
+      final mergeLeft = _BitonicMerge(
+        clk,
+        reset,
+        bitonicSequence: _inputs.getRange(0, _inputs.length ~/ 2),
+        isAscending: isAscending,
+        name: 'merge_left',
+      );
+      final mergeRight = _BitonicMerge(
+        clk,
+        reset,
+        bitonicSequence: _inputs.getRange(_inputs.length ~/ 2, _inputs.length),
+        isAscending: isAscending,
+        name: 'merge_right',
+      );
+
+      final mergeRes = mergeLeft.sorted + mergeRight.sorted;
+
+      for (var i = 0; i < mergeRes.length; i++) {
+        _outputs.add(addOutput('sorted_$i', width: mergeRes[i].width));
+        _outputs[i] <= mergeRes[i];
+      }
+    } else if (_inputs.length == 1) {
+      _outputs.add(addOutput('sorted_0', width: _inputs[0].width));
+      _outputs[0] <= _inputs[0];
+    }
+  }
+}
+
+/// Sort [inputs] to specified order.
+class BitonicSort extends _Sort<BitonicSort> {
+  /// The list of inputs port.
+  final List<Logic> _inputs = [];
+
+  /// The list of outputs port.
+  final List<Logic> _outputs = [];
+
+  /// The [sorted] result.
+  List<Logic> get sorted => UnmodifiableListView(_outputs);
+
+  /// Constructs a [Module] to sort list of [Logic].
+  ///
+  /// The sorting module will recursively split inputs into a bitonic sequence
+  /// and passed to [_BitonicMerge] to perform merging process and return
+  /// the final [sorted] results.
+  BitonicSort(Logic clk, Logic reset,
+      {required super.toSort, super.isAscending, super.name}) {
+    clk = addInput('clk', clk);
+    reset = addInput('reset', reset);
+
+    // Make sure all element have same width
+    int? prevWidth;
+
+    for (final signal in toSort) {
+      prevWidth = prevWidth ?? signal.width;
+
+      if (signal.width != prevWidth) {
+        throw Exception('All inputs width must be the same.');
+      } else {
+        prevWidth = signal.width;
+      }
+    }
+
+    for (var i = 0; i < toSort.length; i++) {
+      _inputs.add(addInput('toSort$i', super.toSort.elementAt(i),
+          width: super.toSort.elementAt(i).width));
+    }
+
+    if (_inputs.length > 1) {
+      final sortLeft = BitonicSort(clk, reset,
+          toSort: _inputs.getRange(0, _inputs.length ~/ 2),
+          name: 'sort_left_${_inputs.length ~/ 2}');
+
+      final sortRight = BitonicSort(clk, reset,
+          toSort: _inputs.getRange(_inputs.length ~/ 2, _inputs.length),
+          isAscending: false,
+          name: 'sort_right_${_inputs.length ~/ 2}');
+
+      final bitonicSequence = sortLeft.sorted + sortRight.sorted;
+
+      final mergeResult = _BitonicMerge(clk, reset,
+              bitonicSequence: bitonicSequence, isAscending: isAscending)
+          .sorted;
+      for (var i = 0; i < mergeResult.length; i++) {
+        _outputs.add(addOutput('sorted_$i', width: _inputs[i].width));
+        _outputs[i] <= mergeResult[i];
+      }
+    } else {
```

Comment:
This will break if the number of elements is 0.  Returning an empty list for `_outputs` is fine if the input list is empty, since that's sorted and removes an edge case for users to consider.

Reply:
The number of elements can't be 0 as the inputs will validate if the inputs is power of 2.